server/middlewares/studentAndTeacher.ts

```diff
@@ -6,6 +6,7 @@ const studentAndTeacher = (req: CustomRequest, res: Response, next: NextFunction
   const { user } = req;
 
   if (user.role === 'teacher' || user.role === 'student') {
+    req.user = user;
```

Comment:
You already passed the `userAuth` middleware where we add user data, so no need for this.